### PR TITLE
Migrated verizon adapter to yahoo mobile sdk.

### PR DESCRIPTION
**Pull request by kristiankuhn**
In order for this adapter to work with AppLovin you will need to create a class com.verizon.ads.VASAds.java in the sample app project so that the AppLovin SDK thinks the Verizon SDK is present. The Yahoo Mobile SDK is what is driving the adapter but the old class is how AppLovin verifies Verizon Ads is library is present so we have to fake it to get the adapter configured successfully.

### CODE REVIEW

**Comment by thomasmso** [comment]
Thanks for the PR - we're testing their new SDK internally and will provide updates when released.

**Comment by kristiankuhn** [comment]
Don't merge this. It's a preview of what is needed to use Yahoo Mobile SDK which is a replacement for Verizon Ads and will be released shortly.

**Comment by thomasmso** [comment]
For sure. Thank you once again for this PR for us to reference.

**Comment by kristiankuhn** [comment]
@thomasmso I've updated the PR to the GA release of Yahoo Mobile SDK. I see there is a merge conflict with the adapter but it's probably not worth resolving as the adapters are completely different between Verizon Ads and Yahoo Mobile. With that said what is your plan for supporting Verizon? Do you intend to support both SDKs for some period of time? In that case we may need to make a new Yahoo adapter and leave the Verizon one as it is. This would apply to both Android and iOS so please advise. Thank you

**Comment by thomasmso** [comment]
We are going to just update the current adapter with the new code.

File: VerizonAds/src/main/java/com/applovin/mediation/adapters/VerizonAdsMediationAdapter.java [hunk 1]
```diff
@@ -335,30 +308,27 @@ public void collectSignal(final MaxAdapterSignalCollectionParameters parameters,
     private void updateVerizonAdsSdkData(final MaxAdapterParameters parameters)
     {
         final int logLevel = parameters.isTesting() ? Logger.VERBOSE : Logger.ERROR;
-        VASAds.setLogLevel( logLevel );
-
-        DataPrivacy.Builder builder = new DataPrivacy.Builder();
+        YASAds.setLogLevel( Logger.VERBOSE );
```
Comment (tiandrew) [inline VerizonAds/src/main/java/com/applovin/mediation/adapters/VerizonAdsMediationAdapter.java:311]: We should use the log level defined above here

File: VerizonAds/src/main/java/com/applovin/mediation/adapters/VerizonAdsMediationAdapter.java [hunk 2]
```diff
@@ -514,8 +465,7 @@ else if ( adFormat == MaxAdFormat.LEADER )
     //endregion
 
     //region Listeners implementation
-    private class InterstitialListener
-            implements InterstitialAdFactory.InterstitialAdFactoryListener, InterstitialAd.InterstitialAdListener
+    private class InterstitialListener implements InterstitialAd.InterstitialAdListener
```
Comment (tiandrew) [inline VerizonAds/src/main/java/com/applovin/mediation/adapters/VerizonAdsMediationAdapter.java:468]: Could you run auto-formatting on this file? `implements` should be on a new line

File: VerizonAds/src/main/java/com/applovin/mediation/adapters/VerizonAdsMediationAdapter.java [hunk 3]
```diff
@@ -704,12 +652,12 @@ private AdViewListener(final MaxAdViewAdapterListener listener)
...
         @Override
-        public void onLoaded(final InlineAdFactory inlineAdFactory, final InlineAdView inlineAdView)
+        public void onLoaded(final InlineAdView inlineAdView)
         {
-            log( "AdView loaded: " + inlineAdFactory.getPlacementId() );
+            log( "AdView loaded: " + inlineAdView.getPlacementId() );
 
             // Disable AdView ad refresh by setting the refresh interval to max value.
-            inlineAdView.setRefreshInterval( Integer.MAX_VALUE );
+//            inlineAdView.setRefreshInterval( Integer.MAX_VALUE );
```
Comment (tiandrew) [inline VerizonAds/src/main/java/com/applovin/mediation/adapters/VerizonAdsMediationAdapter.java:660]: Does this method work still? If so we should probably set it

File: VerizonAds/src/main/java/com/applovin/mediation/adapters/VerizonAdsMediationAdapter.java [hunk 4]
```diff
@@ -820,131 +767,109 @@ public void onLoaded(final NativeAdFactory nativeAdFactory, final NativeAd nativ
...
-                @Override
-                public void run()
+            AppLovinSdkUtils.runOnUiThread(() -> {
+                String title = null;
+                String disclaimer = null;
+                String body = null;
+                String callToAction = null;
+
+                NativeTextComponent titleComponent = (NativeTextComponent)nativeAd.getComponent("title");
+                if (titleComponent != null) {
```
Comment (tiandrew) [inline VerizonAds/src/main/java/com/applovin/mediation/adapters/VerizonAdsMediationAdapter.java:777]: Auto-formatting

File: VerizonAds/src/main/java/com/applovin/mediation/adapters/VerizonAdsMediationAdapter.java [hunk 5]
```diff
@@ -820,131 +767,109 @@ public void onLoaded(final NativeAdFactory nativeAdFactory, final NativeAd nativ
...
+                if ( isTemplateAd && TextUtils.isEmpty( title ) )
+                {
+                    missingRequiredAssets = true;
+                }
+                else if ( TextUtils.isEmpty( title )
+                        || TextUtils.isEmpty( callToAction )
+                        || mediaView == null )
+                {
+                    missingRequiredAssets = true;
+                }
```
Comment (tiandrew) [inline VerizonAds/src/main/java/com/applovin/mediation/adapters/VerizonAdsMediationAdapter.java:828]: We can remove these checks, we're only doing them for template ads now, so you can also just combine this with the case below and remove `missingRequiredAssets`

File: VerizonAds/src/main/java/com/applovin/mediation/adapters/VerizonAdsMediationAdapter.java [hunk 6]
```diff
@@ -820,131 +767,109 @@ public void onLoaded(final NativeAdFactory nativeAdFactory, final NativeAd nativ
...
+                if (ctaComponent != null) {
+                    callToAction = ctaComponent.getText();
+                }
+
+                final NativeImageComponent iconComponent = (NativeImageComponent) nativeAd.getComponent( "iconImage" );
+                ImageView iconView= null;
+                if (iconComponent != null) {
+                    iconView = new ImageView(activity);
+                    iconComponent.prepareView(iconView);
+
```
Comment (tiandrew) [inline VerizonAds/src/main/java/com/applovin/mediation/adapters/VerizonAdsMediationAdapter.java:801]:
```suggestion
```

File: VerizonAds/src/main/java/com/applovin/mediation/adapters/VerizonAdsMediationAdapter.java [hunk 7]
```diff
@@ -987,53 +912,33 @@ public void onEvent(final NativeAd nativeAd, final String source, final String e
...
-                public void run()
-                {
-                    if ( nativeAdView.getTitleTextView() != null ) nativeAdView.getTitleTextView().setOnClickListener( clickListener );
-                    if ( nativeAdView.getAdvertiserTextView() != null ) nativeAdView.getAdvertiserTextView().setOnClickListener( clickListener );
-                    if ( nativeAdView.getBodyTextView() != null ) nativeAdView.getBodyTextView().setOnClickListener( clickListener );
-                    if ( nativeAdView.getIconImageView() != null ) nativeAdView.getIconImageView().setOnClickListener( clickListener );
-                    if ( nativeAdView.getCallToActionButton() != null ) nativeAdView.getCallToActionButton().setOnClickListener( clickListener );
-                }
-            } );
+            nativeAd.registerContainerView(maxNativeAdView);
```
Comment (tiandrew) [inline VerizonAds/src/main/java/com/applovin/mediation/adapters/VerizonAdsMediationAdapter.java:941]: Could you check if multiple native ads using the same recycled `MaxNativeAdView` will allow clicks to work for all the views? We had this issue previously with Verizon's native ads so just want to make sure they fixed it

**Review by tiandrew — CHANGES_REQUESTED** [review]

**Comment by tiandrew** [comment]
Sorry for the confusion, please ignore the requested changes, closing this one; we're iterating on changes internally.

**Comment by thomasmso** [comment]
Adapter is live: https://github.com/AppLovin/AppLovin-MAX-SDK-Android/commit/96c697185bc8a1b5bffc86ac0abb566b2729a9f3

**Pull request closed without merging**